ksz8863/src/ksz8863_ctrl.c

```diff
@@ -57,56 +55,27 @@ static inline bool bus_unlock(void)
 static esp_err_t ksz8863_i2c_write(uint8_t reg_addr, uint8_t *data, size_t len)
 {
     esp_err_t ret = ESP_OK;
-    i2c_port_t i2c_port = s_ksz8863_ctrl_intf->i2c_bus_spec.i2c_port;
-    uint8_t dev_addr = s_ksz8863_ctrl_intf->i2c_bus_spec.dev_addr;
-
-    i2c_cmd_handle_t cmd = i2c_cmd_link_create();
-    ESP_RETURN_ON_FALSE(cmd, ESP_ERR_NO_MEM, TAG, "I2C link create error");
-    ESP_GOTO_ON_ERROR(i2c_master_start(cmd), err, TAG, "I2C master start error");
-    ESP_GOTO_ON_ERROR(i2c_master_write_byte(cmd, dev_addr | I2C_MASTER_WRITE, true), err, TAG, "I2C master write error");
-    ESP_GOTO_ON_ERROR(i2c_master_write_byte(cmd, reg_addr, true), err, TAG, "I2C master write error");
-    if (likely(reg_addr != KSZ8863_RESET_ADDR)) {
-        ESP_GOTO_ON_ERROR(i2c_master_write(cmd, data, len, true), err, TAG, "I2C master write error");
+    uint8_t *reg_addr_and_data = malloc(len + 1);
+    // Create a packet containing the register and data to be transmitted
+    reg_addr_and_data[0] = reg_addr;
+    memcpy(reg_addr_and_data + 1, data, len);
+    // When performing soft reset KSZ8863 doesn't produce ACK. Print a wanrning that the error is expected and ignore it.
+    if unlikely(reg_addr == KSZ8863_RESET_ADDR) {
+        ESP_LOGW(TAG, "Ignore the error below. KSZ8863 does not produce ACK when performing soft reset.");
+        i2c_master_transmit(s_ksz8863_ctrl_intf->i2c_handle, reg_addr_and_data, len + 1, KSZ8863_I2C_TIMEOUT_MS);
     } else {
-        // when SW reset is performed, KSZ does not generate ACK
-        ESP_GOTO_ON_ERROR(i2c_master_write(cmd, data, len, false), err, TAG, "I2C master write error");
+        ESP_GOTO_ON_ERROR(i2c_master_transmit(s_ksz8863_ctrl_intf->i2c_handle, reg_addr_and_data, len + 1, KSZ8863_I2C_TIMEOUT_MS), err, TAG, "Error during i2c write operation");
     }
-    ESP_GOTO_ON_ERROR(i2c_master_stop(cmd), err, TAG, "I2C master stop error");
-    // Lock since multiple MAC/PHY instances exist and the KSZ may be also accessed by user (MAC tables, etc...)
-    ESP_GOTO_ON_FALSE(bus_lock(KSZ8863_I2C_LOCK_TIMEOUT_MS), ESP_ERR_TIMEOUT, err, TAG, "I2C bus lock timeout");
```

Comment:
Mutex (`s_ksz8863_ctrl_intf->bus_lock` ) is still created for even when I2C mode is used.

Reply:
Moved to `case KSZ8863_SPI_MODE`

Reply:
do the same in `ksz8863_ctrl_intf_deinit`